Using `@pnpm/filter-workspace-packages` from executable script results in "Unsupported engine" stating current pnpm is "0.0.0"

I'd like to use `@pnpm/filter-workspace-packages` to programmatically explore dependencies. The package failed until I added `@pnpm/logger`. At this point the below error occurs when I use the package in a script.

### pnpm version:
```
$ pnpm --version
6.10.3
```

### Code to reproduce the issue:
```
$ cat list-files.mjs
#!/usr/bin/env node
import { readProjects } from '@pnpm/filter-workspace-packages'
console.log(await readProjects(process.cwd(), [], { linkWorkspacePackages: true }))
$ ls -lha list-files.mjs 
.rwxr-xr-x  3.1k jon  29 Aug 10:02  list-files.mjs
$ ./list-files.mjs
<path>/node_modules/.pnpm/@pnpm+package-is-installable@5.0.4_@pnpm+logger@4.0.0/node_modules/@pnpm/package-is-installable/lib/checkEngine.js:29
        return new UnsupportedEngineError(packageId, unsatisfiedWanted, currentEngine);
               ^

UnsupportedEngineError: Unsupported engine for <path>: wanted: {"pnpm":">= 6.9.0"} (current: {"node":"v16.1.0","pnpm":"0.0.0"})
    at Object.checkEngine [as default] (<path>/node_modules/.pnpm/@pnpm+package-is-installable@5.0.4_@pnpm+logger@4.0.0/node_modules/@pnpm/package-is-installable/lib/checkEngine.js:29:16)
    at Object.checkPackage (<path>/node_modules/.pnpm/@pnpm+package-is-installable@5.0.4_@pnpm+logger@4.0.0/node_modules/@pnpm/package-is-installable/lib/index.js:61:32)
    at Object.packageIsInstallable (<path>/node_modules/.pnpm/@pnpm+cli-utils@0.6.17_@pnpm+logger@4.0.0/node_modules/@pnpm/cli-utils/lib/packageIsInstallable.js:15:42)
    at Object.exports.default (<path>/node_modules/.pnpm/@pnpm+find-workspace-packages@3.1.9_@pnpm+logger@4.0.0/node_modules/@pnpm/find-workspace-packages/lib/index.js:15:21)
    at async readProjects (<path>/node_modules/.pnpm/@pnpm+filter-workspace-packages@4.1.11_@pnpm+logger@4.0.0/node_modules/@pnpm/filter-workspace-packages/lib/index.js:38:25)
    at async file://<path>/list-files.mjs:3:20 {
  code: 'ERR_PNPM_UNSUPPORTED_ENGINE',
  hint: undefined,
  attempts: undefined,
  packageId: '<path>',
  wanted: { pnpm: '>= 6.9.0' },
  current: { node: 'v16.1.0', pnpm: '0.0.0' }
}
```
(I've replaced my folder path with "<path>" to shorten the output)

### Expected behavior:
The correct version of pnpm should be detected, or at least a way to bypass, force, or inform the system of the actual pnpm version.

### Actual behavior:
pnpm is identified as version "0.0.0" which trips the engine requirement.

### Additional information:
```
$ node -v
v16.1.0
```

macOS

--- a/.changeset/modern-gifts-end.md
+++ b/.changeset/modern-gifts-end.md
@@ -1,0 +1,5 @@
+---
+"@pnpm/filter-workspace-packages": minor
+---
+
+New optional option added to `readProjects()`: engineStrict.

--- a/packages/filter-workspace-packages/src/getChangedPackages.ts
+++ b/packages/filter-workspace-packages/src/getChangedPackages.ts
@@ -67,7 +67,7 @@ async function getChangedDirsSinceCommit (commit: string, workingDir: string, te
   const patterns = changedFilesIgnorePattern.filter(
     (pattern) => pattern.length
   )
-  const changedFiles = patterns.length
+  const changedFiles = (patterns.length > 0)
     ? micromatch.not(allChangedFiles, patterns, {
       dot: true,
     })

--- a/packages/filter-workspace-packages/src/index.ts
+++ b/packages/filter-workspace-packages/src/index.ts
@@ -33,11 +33,12 @@ export async function readProjects (
   workspaceDir: string,
   pkgSelectors: PackageSelector[],
   opts?: {
+    engineStrict?: boolean
     linkWorkspacePackages?: boolean
     changedFilesIgnorePattern?: string[]
   }
 ) {
-  const allProjects = await findWorkspacePackages(workspaceDir, {})
+  const allProjects = await findWorkspacePackages(workspaceDir, { engineStrict: opts?.engineStrict })
   const { selectedProjectsGraph } = await filterPkgsBySelectorObjects(
     allProjects,
     pkgSelectors,

